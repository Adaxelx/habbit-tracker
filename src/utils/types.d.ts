export type GridProps = {
  events: Event[];
  month: number;
  year: number;
  moveDate: Function;
};

export interface Event {
  _id: string;
  daysOfWeek: number[];
  title: string;
  timeStart?: string;
  timeEnd?: string;
  dateStart: string;
  dateEnd: string;
  label?: string | Label;
  description?: string;
  userId: string;
  checked?: boolean;
}

export interface EventLoop extends Event {
  numericStart: number;
  numericEnd: number;
}

export type Label = {
  _id: string;
  title: string;
  color: string;
};

export type CalendarTile = {
  day: number;
  events: EventLoop[];
  id: number;
};

export enum SIDES {
  LEFT = -1,
  RIGHT = 1,
}
