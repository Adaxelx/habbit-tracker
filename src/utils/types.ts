export type GridProps = {
  events: Event[];
  month: number;
  year: number;
  moveDate: Function;
  handleDayChange: Function;
};

export interface Event {
  _id: string;
  daysOfWeek: number[];
  title: string;
  timeStart?: string;
  timeEnd?: string;
  dateStart: string;
  dateEnd: string;
  label?: string | Label;
  description?: string;
  userId: string;
  checked?: [
    {
      day: number;
      month: number;
      year: number;
      _id: string;
    },
  ];
}

export interface EventLoop extends Event {
  numericStart: number;
  numericEnd: number;
}

export type Label = {
  _id: string;
  title: string;
  color: string;
  userId: string;
};

export type CalendarTile = {
  day: number;
  events: EventLoop[];
  id: number;
};

export enum SIDES {
  LEFT = -1,
  RIGHT = 1,
}

export type Color = { color?: string; id?: string };

export type NavProps = {
  header: string;
  moveDate: Function;
};

export interface DayCardProps {
  day: string;
  token: string | undefined;
}
