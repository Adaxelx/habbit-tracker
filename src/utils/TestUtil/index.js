export { default } from './TestUtil.ts';
