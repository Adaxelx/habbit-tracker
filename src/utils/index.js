export { default as TestUtil } from './TestUtil';
export { default as RenderWithRouter } from './RenderWithRouter';
export * from './validators';
export * from './types.d';
