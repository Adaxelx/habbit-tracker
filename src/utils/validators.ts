export const createRequiredObject = (name: string) => ({
  required: `Field ${name} is required.`,
});

export const validatePattern = (pattern: RegExp) => ({
  pattern: {
    value: pattern,
    message: 'Incorrect characters.',
  },
});

export const emailValidation = validatePattern(
  // eslint-disable-next-line no-control-regex
  /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/,
);

export const createRestrictedLengthObject = (name: string) => ({
  ...createRequiredObject(name),
  minLength: {
    value: 6,
    message: `Field ${name} must have at least 6 characters.`,
  },
  maxLength: {
    value: 30,
    message: `Field ${name} must have maximum 30 characters.`,
  },
});

const timeValidFuntion = (timeLess: string | undefined, time: string | undefined) => {
  const message = 'Your time is not in boundaries.';
  if (timeLess && time) {
    const [hours, minutes] = time.split(':');
    const [hoursMin, minutesMin] = timeLess.split(':');
    if (hoursMin < hours) {
      return true;
    }
    if (hoursMin === hours) {
      if (minutesMin < minutes) {
        return true;
      }
      return message;
    }
    return message;
  }
  return message;
};

export const createTimeValidation = (min: string | undefined, max: string | undefined) => ({
  validate: {
    min: (value: string) => timeValidFuntion(min, value),
    max: (value: string) => timeValidFuntion(value, max),
  },
});
