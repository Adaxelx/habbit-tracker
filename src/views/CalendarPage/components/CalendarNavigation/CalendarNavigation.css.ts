import styled from 'styled-components';

export const StyledNavigation = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin: ${({ theme }) => theme.margin.s} 0;
`;

export const StyledButton = styled.button`
  background-color: transparent;
  border: none;
  font-size: ${({ theme }) => theme.font.sizes.m};
`;

export const StyledTitle = styled.h3`
  width: 200px;
  text-align: center;
  line-height: 100%;
  margin: 0;
  @media (min-width: ${({ theme }) => theme.media.tablet.s}) {
    width: 400px;
    font-size: ${({ theme }) => theme.font.sizes.ml};
  }
`;
