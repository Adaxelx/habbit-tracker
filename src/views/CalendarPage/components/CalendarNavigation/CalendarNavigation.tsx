import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCaretLeft, faCaretRight } from '@fortawesome/free-solid-svg-icons';
import { months } from 'constants/calendar';
import { SIDES, NavProps } from 'utils';
import { StyledNavigation, StyledButton, StyledTitle } from './CalendarNavigation.css';

const CalendarNavigation = ({ month, year, moveDate }: NavProps) => (
  <StyledNavigation>
    <StyledButton onClick={() => moveDate(SIDES.LEFT)}>
      <FontAwesomeIcon icon={faCaretLeft} />
    </StyledButton>
    <StyledTitle>{`${months[month]} ${year}`}</StyledTitle>
    <StyledButton onClick={() => moveDate(SIDES.RIGHT)}>
      <FontAwesomeIcon icon={faCaretRight} />
    </StyledButton>
  </StyledNavigation>
);

export default CalendarNavigation;
