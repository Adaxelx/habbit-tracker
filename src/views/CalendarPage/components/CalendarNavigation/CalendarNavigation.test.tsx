import React from 'react';
import 'jest-styled-components';
import { TestUtil, NavProps } from 'utils';
import CalendarNavigation from './CalendarNavigation';

describe('CalendarNavigation', () => {
  let util: TestUtil;

  let props: NavProps;

  beforeEach(() => {
    jest.clearAllMocks();

    props = {
      header: 'April 2021',
      moveDate: jest.fn(),
    };
    util = new TestUtil(<CalendarNavigation {...props} />);
  });

  it('should match snapshot', () => {
    expect(util.render.asFragment()).toMatchSnapshot();
  });
});
