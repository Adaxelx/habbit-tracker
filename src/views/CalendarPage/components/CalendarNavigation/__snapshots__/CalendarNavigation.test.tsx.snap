// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarNavigation should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0.75rem;
}

.c1 {
  background-color: transparent;
  border: none;
  font-size: 1.5rem;
}

.c2 {
  width: 200px;
  text-align: center;
  line-height: 100%;
  margin: 0;
}

@media (min-width:768px) {
  .c2 {
    width: 400px;
    font-size: 2rem;
  }
}

<div
    class="c0"
  >
    <button
      class="c1"
      data-testid="moveLeft"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-caret-left fa-w-6 "
        data-icon="caret-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 192 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
          fill="currentColor"
        />
      </svg>
    </button>
    <h3
      class="c2"
      data-testid="dateCalendar"
    >
      April 2021
    </h3>
    <button
      class="c1"
      data-testid="moveRight"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-caret-right fa-w-6 "
        data-icon="caret-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 192 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
